tests/TestCase/Http/Middleware/CsrfProtectionMiddlewareTest.php

```diff
@@ -100,6 +100,48 @@ public function testSettingCookie()
         $this->assertEquals($cookie['value'], $updatedRequest->getAttribute('csrfToken'));
     }
 
+    /**
+     * Test invalid cookie is replaced with valid cookie
+     *
+     * @return void
+     */
+    public function testInvalidCookieTokenRegenerated()
+    {
+        $request = new ServerRequest([
+            'environment' => ['REQUEST_METHOD' => 'GET'],
+            'cookies' => ['csrfToken' => 'invalidToken'],
```

Comment:
Should we have a test that covers the arbitrary cookie data working when it is submitted with an hmac'ed value?

Reply:
There are few tests which uses hmac'ed value (testValidTokenInHeader, testValidTokenRequestData) so I think no extra test necessarily.